airqo-platform/AirQo-api
# Ft join platform

- What does this PR do?
This PR creates refactors code to take into account some new user access needs. At the same time, it introduces some new exciting endpoints to cater for the different user needs. Schemas are also updated accordingly to reflect the needs of the new endpoints.
- What JIRA task is related to this PR?
[Refactoring the auth microservice.](https://airqoteam.atlassian.net/browse/NLYTCS-67)
- How do I test out this PR?
Git clone this repo `cd AirQo-api/src/auth-service` Then run `npm install` to get all the necessary dependencies Afterwards run `npm run dev` to test out the end points using any REST client tool available to you.
- Which end points should are ready for testing?
base URI: http://localhost:3000/
partial url: /api/v1/forecast/feeds/recent/<CHANNEL_ID>

1. [POST] partial url: api/v1/users/registerUser
```
{
  "firstName": "<STRING>",
  "lastName": "<STRING>",
  "email": "<EMAIL>",
  "userName": "<STRING>",
  "password": "<PASSWORD>",
  "password2": "<PASSWORD>"
}
```
2. [GET] partial url: api/v1/users
3. [POST] partial url: api/v1/users/registerCandidate
4. [POST] partial url: api/v1/users/activateCandidate/:_USER ID_
5. [GET] partial url: api/v1/users/getOneCandidate/:_USER ID_
6. [GET] partial url: api/v1/users/getCandidates
7. [POST] partial url: api/v1/users/registerCollab
8. [PUT] partial url: api/v1/users/updateCollab/:_USER_ID_
9. [DELETE] partial url: api/v1/users/deleteCollab/:_USER_ID_
10. [GET] partial url: api/v1/users/collaborators

For more specific details, please head over to the [AIRQO API documentation](https://docs.google.com/spreadsheets/d/1-NGTDBtTcIi8Fg2UYSWjcK6geKrZdopz5kxRFsCEMPs/edit?usp=sharing)
